generators/generator-adaptive-bot/generators/app/templates/dotnet/functions/Startup.cs

```diff
@@ -11,21 +14,27 @@ public class Startup : FunctionsStartup
         public override void Configure(IFunctionsHostBuilder builder)
         {
             builder.Services.AddBotRuntime(builder.GetContext().Configuration);
+
+            // The workaround below will be removed next SDK patch, when this bug fix gets released: https://github.com/microsoft/botbuilder-dotnet/issues/5239
+            // In the meantime, we're guaranteed to have CoreAdapter registered as IBotFrameworkHttpAdapter, so look it up and register it as BotAdapter.
+            RegisterCoreBotAdapter(builder.Services);
         }
 
         public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder configurationBuilder)
         {
             string applicationRoot = configurationBuilder.GetContext().ApplicationRootPath;
             string settingsDirectory = <%- settingsDirectory %>;
-            bool isDevelopment = string.Equals(
-                configurationBuilder.GetContext().EnvironmentName,
-                Microsoft.Extensions.Hosting.Environments.Development,
-                StringComparison.OrdinalIgnoreCase);
 
-            configurationBuilder.ConfigurationBuilder.AddBotRuntimeConfiguration(
-                applicationRoot,
-                settingsDirectory,
-                isDevelopment);
+            configurationBuilder.ConfigurationBuilder.AddBotRuntimeConfiguration(applicationRoot, settingsDirectory);
+        }
+
+        private static BotAdapter RegisterCoreBotAdapter(IServiceCollection services)
+        {
```

Comment:
This cruft should go away as soon as we add a new RC build or we'll do a preview patch right after Rc0 ships.